providers/class.two-factor-email.php

```diff
@@ -73,10 +73,15 @@ public function generate_token( $user_id ) {
 	 */
 	public function validate_token( $user_id, $token ) {
 		$hashed_token = get_user_meta( $user_id, self::TOKEN_META_KEY, true );
-		if ( wp_hash( $token ) !== $hashed_token ) {
-			$this->delete_token( $user_id );
+
+		// Bail if token is empty or it doesn't match.
+		if ( empty( $hashed_token ) || ( wp_hash( $token ) !== $hashed_token ) ) {
```

Comment:
Shouldn't this return false if `$hashed_token` is empty?  If the meta somehow gets otherwise deleted, we probably shouldn't treat it as a successful match?